packages/financial-templates-lib/src/price-feed/FundingRateMultiplierPriceFeed.js

```diff
@@ -0,0 +1,105 @@
+const { PriceFeedInterface } = require("./PriceFeedInterface");
+const { BlockFinder } = require("./utils");
+const { ConvertDecimals } = require("@uma/common");
+const { aggregateTransactionsAndCall } = require("../helpers/multicall");
+const assert = require("assert");
+
+class FundingRateMultiplierPriceFeed extends PriceFeedInterface {
+  /**
+   * @notice Constructs new price feed object that tracks the funding rate multiplier in a particular perpetual contract.
+   * @param {Object} logger Winston module used to send logs.
+   * @param {Object} perpetualAbi Perpetual abi object to create a contract instance.
+   * @param {Object} web3 Provider from Truffle instance to connect to Ethereum network.
+   * @param {String} perpetualAddress Ethereum address of the perpetual to monitor.
+   * @param {String} multicallAddress Ethereum address of the multicall contract.
+   * @param {Function} getTime Returns the current time.
+   * @param {Function} [blockFinder] Optionally pass in a shared blockFinder instance (to share the cache).
+   * @param {Integer} [minTimeBetweenUpdates] Minimum time that must pass before update will actually run again.
+   * @param {Integer} [priceFeedDecimals] Precision that the caller wants precision to be reported in.
+   * @return None or throws an Error.
+   */
+  constructor({
+    logger,
+    perpetualAbi,
+    web3,
+    perpetualAddress,
+    multicallAddress,
+    getTime,
+    blockFinder,
+    minTimeBetweenUpdates = 60,
+    priceFeedDecimals = 18
+  }) {
+    super();
+
+    // Assert required inputs.
+    assert(logger, "logger required");
+    assert(perpetualAbi, "perpetualAbi required");
+    assert(web3, "web3 required");
+    assert(web3.utils.isAddress(multicallAddress), "multicallAddress required");
+    assert(web3.utils.isAddress(perpetualAddress), "perpetualAddress required");
+    assert(getTime, "getTime required");
+
+    this.logger = logger;
+    this.web3 = web3;
+
+    this.perpetual = new web3.eth.Contract(perpetualAbi, perpetualAddress);
+    this.multicallAddress = multicallAddress;
+    this.uuid = `FundingRateMultiplier-${perpetualAddress}`;
+    this.getTime = getTime;
+    this.priceFeedDecimals = priceFeedDecimals;
+    this.minTimeBetweenUpdates = minTimeBetweenUpdates;
+    this.blockFinder = blockFinder || BlockFinder(web3.eth.getBlock);
+    this.convertDecimals = ConvertDecimals(18, priceFeedDecimals, web3);
+  }
+
+  getCurrentPrice() {
+    return this.price;
+  }
+
+  async getHistoricalPrice(time) {
+    const block = await this.blockFinder.getBlockForTimestamp(time);
+    return this._getPrice(block.number);
```

Comment:
what if `block == undefined`?

Reply:
Then that should be an error, right?

Reply:
I don't think the blockfinder returns undefined. It will error out if it cannot find a block.